src/module/system/damage/damage.ts

```diff
@@ -20,7 +20,8 @@ export class DamagePF2e {
     ): Promise<Rolled<DamageRoll> | null> {
         const outcome = context.outcome ?? null;
 
-        context.rollMode ??= (context.secret ? "blindroll" : undefined) ?? game.settings.get("core", "rollMode");
+        context.rollMode = context.secret ? "blindroll" : context.rollMode ?? game.settings.get("core", "rollMode");
```

Comment:
A few things regarding this:
1) If rollMode is supplied, we need to always accept it. If someone does a public secret roll, then by golly, they want a public secret roll. This means the first ??= is correct. 
2) Please put parenthesis. The order of operations for ?? is unclear without them.
3) The space after this probably shouldn't exist. The reason context.rollMode and context.createMessage were together with no gaps between is that both pieces of code involve setting default users if the user didn't supply data.

If this isn't the bugfix, and the bugfix involves what's in damage/dialog.ts, it might be best to omit this change.